ggshield/auth.py

```diff
@@ -10,21 +13,65 @@
     type=click.Choice(["token"]),
     help="Authentication method.",
 )
+@click.option(
+    "--instance",
+    required=False,
+    type=str,
+    help="URL of the instance to authenticate to.",
+)
 @click.pass_context
-def login_cmd(ctx: click.Context, method: str) -> int:
+def login_cmd(ctx: click.Context, method: str, instance: str) -> int:
     """Authenticate to your GitGuardian account."""
+    config = ctx.obj["config"]
+
+    if instance:
+        if not instance.startswith("https://"):
+            raise click.BadParameter("Instance must be a valid https URL.")
+
+        # Normalize instance name by removing the path, if it is set
+        parsed_url = urlparse(instance)
+        instance = f"https://{parsed_url.netloc}"
```

Comment:
nit: maybe you can reuse the `clean_url` method I added in [this PR](https://github.com/GitGuardian/ggshield/pull/198)

Reply:
Ok I'll wait for your PR to be merged then

Reply:
I just merged Eugène PR, you can now use `clean_url`.